src/core/public/ui_settings/ui_settings_client.ts

```diff
@@ -230,8 +268,15 @@ You can use \`IUiSettingsClient.get("${key}", defaultValue)\`, which will just r
 
     const declared = this.isDeclared(key);
     const defaults = this.defaults;
-
-    const oldVal = declared ? this.cache[key].userValue : undefined;
+    const userControl =
+      this.cache['theme:enableUserControl']?.userValue ??
```

Comment:
ui_settings_client shouldn't be aware of a theme specific setting right?

also, could other settings be null? if so, this probably should be checking that these aren't defined.